tools/topic_tools/include/topic_tools/shape_shifter.h

```diff
@@ -222,12 +222,8 @@ void ShapeShifter::write(Stream& stream) const {
 template<typename Stream>
 void ShapeShifter::read(Stream& stream)
 {
-  stream.getLength();
-  stream.getData();
-
   // stash this message in our buffer
-  msgBuf.resize(stream.getLength());
-  memcpy(msgBuf.data(), stream.getData(), stream.getLength());
+  msgBuf.insert(msgBuf.end(), stream.getData(), stream.getData() + stream.getLength());
```

Comment:
If I understand correctly, the original code is replacing the content of `msgBuf` with the content of the stream, while this new code appends the stream data to the end of msgBuf.

Maybe `assign` would be a better choice?

Reply:
If the internal implementation of assign does not do a memset, then yes :)
When I originally implemented this, we were still using c++17 and assign was not available.

Maybe check the compiler explorer to be sure that it does not do memset